Sources/MediaCategories/MediaCategoryViewController.swift

```diff
@@ -235,23 +205,67 @@ class VLCMediaCategoryViewController: UICollectionViewController, UICollectionVi
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
         return category.cellType.cellPadding
     }
-}
 
-// MARK: - Sort
+    override func handleSort() {
+        let sortOptionsAlertController = UIAlertController(title: NSLocalizedString("SORT_BY", comment: ""),
+                                                           message: nil,
+                                                           preferredStyle: .actionSheet)
 
-extension VLCMediaCategoryViewController {
-    // FIXME: Need to add a button for ascending/descending result
-    func sortByFileName() {
-        // The issue is that for audio we show title which is quite confusing if we use filename
-        category.sort(by: .alpha)
+        var alertActions = [UIAlertAction]()
+
+        for (index, enabled) in category.sortModel.sortingCriteria.enumerated() {
+            guard enabled else { continue }
+            let criteria = VLCMLSortingCriteria(value: UInt(index))
+
+            alertActions.append(UIAlertAction(title: criteria.description, style: .default) {
+                [weak self] action in
+                self?.category.sortModel.currentSort = criteria
+                self?.category.sort(by: criteria)
```

Comment:
How do you feel about changing the currentSort automatically triggers the sort? If you have the above you could theoretically set a currentSort to one thing and then sort by something else

Reply:
You're totally right, I believe this should be inside `sort()`.